modules/_labs/popup-stack/react/lib/PopupStack.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * This type is purposely an interface so that it can be extended for a
+ * specific use-case
+ */
+export interface PopupStackItem {
+  /**
+   * Ephemeral type popups usually have global listeners like click outside
+   * or escape key handling. Ephemeral type popups include Tooltips, Modals,
+   * DropDowns, etc. Persistent type popups include Windows. Persistent can
+   * only be dismissed explicitly by a close button.
+   */
+  type: 'persistent' | 'ephemeral';
+  element: HTMLElement;
+}
+
+function getLast<T>(items: T[]): T | null {
+  if (items.length) {
+    return items[items.length - 1];
+  }
+  return null;
+}
+
+/**
+ * Set the z-index of each item in the stack by changing setting
+ * `element.style.zIndex`. This should work for a web target across all
+ * frameworks. The strategy
+ */
+export function defaultGetValue(index: number, length: number): number {
+  const {min, max} = stack.zIndex;
+
+  return Math.max(min, max + index - length + 1);
+}
```

Comment:
Consider changing z-index assignment strategy to not change z-index of all other popups